src/main/java/com/ericsson/ei/frontend/utils/BackendInstancesHandler.java

```diff
@@ -0,0 +1,143 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.ericsson.ei.frontend.exceptions.EiBackendInstancesException;
+import com.ericsson.ei.frontend.model.BackendInstance;
+import com.google.gson.JsonArray;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@Component
+public class BackendInstancesHandler {
+    private static final Logger LOG = LoggerFactory.getLogger(BackendInstancesHandler.class);
+
+    @Value("${ei.backend.instances.list.json.content:#{null}}")
+    private String backendInstancesJsonString;
+    private List<BackendInstance> backendInstances = new ArrayList<>();
+
+    @PostConstruct
+    private void init() {
+        parseBackendInstancesStringToBackendInstances();
+    }
+
+    /**
+     * Returns the backendInstances as JsonArray
+     * @return JsonArray
+     */
+    public JsonArray getBackendInstancesAsJsonArray() {
+        JsonArray backendJsonArray = new JsonArray();
+        for (BackendInstance backendInstance : backendInstances) {
+            backendJsonArray.add(backendInstance.getAsJsonObject());
+        }
+        return backendJsonArray;
+    }
+
+    /**
+     * Returns the wanted BackendInstance
+     *
+     * @param backendName
+     * @return BackendInstance
+     * @throws EiBackendInstancesException
+     */
+    public BackendInstance getBackendInstance(String backendName) throws EiBackendInstancesException {
+        for(BackendInstance backendInstance: backendInstances) {
+            if(backendInstance.getName().equals(backendName)) {
+                LOG.debug("Returning backend " + backendName + ".");
+                return backendInstance;
+            }
+        }
+
+        throw new EiBackendInstancesException(
+                "EI Backend instance '" + backendName +  "' does not exist in EI Backend Instances list.");
+        }
+
+    /**
+     * Returns the default backend if set, otherwise it will return
+     * the first instance in the list.
+     *
+     * @return BackendInstance
+     */
+    public BackendInstance getDefaultBackendInstance() {
+        for(BackendInstance backendInstance: backendInstances) {
+            if(backendInstance.isDefaultBackend()) {
+                return backendInstance;
+            }
+        }
+
+        return backendInstances.get(0);
+    }
+
+    /**
+     * Sets the backendInstances list, useful when mocking for tests
+     *
+     * @param backendInstances
+     */
+    public void setBackendInstances(JsonArray backendInstances) {
+        cleanBackendInstances();
+        backendInstancesJsonString = backendInstances.toString();
```

Comment:
Do backendInstancesJsonString has to be a global variable or is it only used here and in parse String?, One may send the parsed string to 'parseBackendInstancesStringToBackendInstances'.

The function parseBackendInstancesStringToBackendInstances could be named 'setBackendInstances' as it is what it does, but instead recieve a string.

Heck it actually convers a JsonArray to String, then converts the String to a JsonArray...

Reply:
backendInstancesJsonString is read from application.properties so it has to be a private variable.

I don't quite agree with the reasoning behind parseBackendInstancesStringToBackendInstances. There is already a method named setBackendInstances, that method has a helper method to parse the instances from the private variable of instances that has string format to the BackendInstances object, that is it's main objective. Then the helper method itself is doing stuff within it yes but the main objective is to parse from a structure into another.